### DISCUSSION TITLE
How to Mock a Firebase Function?

### DISCUSSION THREAD

**Discussion by jelling** [storybookjs/storybook#21848] [Help]
Can anyone show me how to mock a Firebase function inside of Storybook? 

Function I'm trying to mock:
```
import { getDownloadURL } from "firebase/storage";
```

Tried using `@storybook/jest` like this but the function isn't mocked when the Story calls it:

```
import { jest } from "@storybook/jest";
jest.fn(getDownloadURL).mockResolvedValue("https://example.com/image.png");
jest.fn(getDownloadURL).mockImplementation(() => Promise.resolve("foo"));
```

Tried getting Jest this way but not allowed: 
```
import { jest } from "@jest/globals";
```

And this caused all sorts of missing poly-fill errors:

```
const jest = require("jest");
jest.mock("firebase/storage", () => ({
  ...jest.requireActual("firebase/storage"),
  getDownloadURL: jest.fn(() =>
    Promise.resolve("https://example.com/image.png")
  ),
}));
```

[storybook-addon-module-mock](https://github.com/ReactLibraries/storybook-addon-module-mock) seems promising but it's for mocking entire modules and I couldn't get it work, either.

**Comment by buggedcom** [comment]
Did you find a suitable way to mock?
